model.py

```diff
@@ -0,0 +1,179 @@
+import numpy as np
+from sklearn.dummy import DummyClassifier
+import seaborn as sns
+import matplotlib.pyplot as plt
+from sklearn.preprocessing import StandardScaler
+from sklearn.decomposition import PCA
+from sklearn.linear_model import LogisticRegression
+from sklearn.neighbors import KNeighborsClassifier
+from sklearn.pipeline import Pipeline
+from sklearn.model_selection import GridSearchCV
+from sklearn.svm import SVC
+from sklearn import svm
+from imblearn.over_sampling import SMOTE
+from sklearn.neighbors import KNeighborsClassifier
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.ensemble import RandomForestClassifier, AdaBoostClassifier
+import io
+
+from google.colab import files
+uploaded = files.upload()
+
+#import train dataset
+X = pd.read_csv(io.BytesIO(uploaded['train_data.csv']))
+
+#import labels
+y = pd.read_csv(io.BytesIO(uploaded['train_labels.csv']))
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
+
+#y = y['y'].apply(lambda x: 1 if x == -1 else 0)
+
+
+#baseline
+strategies = ['stratified', 'most_frequent', 'prior', 'uniform']
+
+for i in strategies:
+    dummy_clf = DummyClassifier(strategy=i)
+    dummy_clf.fit(X_train, y_train)
+    score = dummy_clf.score(X_train, y_train)
+    dum_labels = np.unique(dummy_clf.predict(X))
+    print(f'For {i} strategy score is {score}')
+    plt.bar(i, score)
+
+
+#scaling data
+
+scaler = StandardScaler()
+
+X_train_scale = scaler.fit_transform(X_train)
+y_train_scale = scaler.fit_transform(y_train)
+
+
+#PCA
+
+pca = PCA(n_components=2, whiten=True)
+X_pca = pca.fit_transform(X_train_scale)
+
+
+#Logistic Regression
+
+clf = LogisticRegression(random_state=0).fit(X, y)
+
+#KNeighborsClassifier
+
+knn = KNeighborsClassifier().fit(X, y)
+
+#SVC
+
+svc = svm.SVC()
+svc.fit(X, y)
+
+
+#RandomForestClassifier
+
+rnf = RandomForestClassifier().fit(X, y)
+
+
+#DecisionTreeClassifier
+
+dtc = DecisionTreeClassifier().fit(X, y)
+
+
+#creating pipeline
+
+pipe = Pipeline([('pca', pca), ('scaler', scaler), 
+                ('classifier', SVC())
+            ])
+
+
+search_space = [{'scaler': [StandardScaler()],
+                'pca': [PCA()]},
+                {'classifier': [SVC()],
+                 'classifier__kernel': ['linear', 'rbf', 'poly'],
+                 'classifier__class_weight': [None, 'balanced'],
+                 'classifier__gamma': ['scale', 'auto'],
+                 'classifier__C': np.logspace(1,4,10)},
+                {'classifier': [KNeighborsClassifier()],
+                 'classifier__n_neighbors': [2, 4, 6, 8, 10, 20],
+                 'classifier__algorithm': ['auto']}]
+
+grid_search = GridSearchCV(pipe,
+                           search_space,
+                           cv=5,
+                           verbose=2,
+                           n_jobs=-2,
+                           scoring='balanced_accuracy')
+
+best_model = grid_search.fit(X, y.values)
+
+
+#choosing best model
```

Comment:
Personally, I really like separating code with lines in the form of a dash. For example: 
`#-----------------------------------------------------`
Then you can see exactly where the new part starts.